packages/playwright-core/src/server/trace/recorder/tracing.ts

```diff
@@ -194,6 +198,44 @@ export class Tracing extends SdkObject implements InstrumentationListener, Snaps
     return { traceName: this._state.traceName };
   }
 
+  async group(name: string, options: { location?: { file: string, line?: number, column?: number } } = {}): Promise<void> {
+    if (!this._state)
+      return;
+    const stackFrame: StackFrame = {
+      file: options.location?.file || '',
+      line: options.location?.line || 0,
+      column: options.location?.column || 0,
+    };
+    const event: trace.BeforeActionTraceEvent = {
+      type: 'before',
+      callId: `group-${this._state.groupId++}`,
```

Comment:
I'd reuse the `metadata.id` from the `metadata` that is passed to `TracingDispatcher.tracingGroup()` as a second parameter. That will connect stacks/sources machinery to this trace event automatically.

Reply:
OK. WOW!

i was totally unaware of the metadata. 
It does not only solve the `callId` but also the `location` as `stackFrame` and the `startTime`

can you please confirm that we can use the metadata as i did now?